clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/HonoKafkaConsumer.java

```diff
@@ -238,7 +262,18 @@ void setKafkaConsumerSupplier(final Supplier<Consumer<String, Buffer>> supplier)
             kafkaConsumer = Optional.ofNullable(kafkaConsumerSupplier)
                     .map(supplier -> KafkaConsumer.create(vertx, supplier.get()))
                     .orElseGet(() -> KafkaConsumer.create(vertx, consumerConfig, String.class, Buffer.class));
-            kafkaConsumer.handler(recordHandler);
+            kafkaConsumer.handler(record -> {
+                if (respectTtl && KafkaRecordHelper.isTtlElapsed(record.headers())) {
+                    onRecordHandlerSkippedForExpiredRecord(record);
+                } else {
+                    try {
+                        recordHandler.handle(record);
+                    } catch (final Exception e) {
+                        log.warn("error handling record [topic: {}, partition: {}, offset: {}, headers: {}]",
```

Comment:
FMPOV skipping a record in an _at least once_ scenario is an essential point. It would be good to provide more options to react on that, e.g. put it into the traces.

Reply:
This try-catch block is really meant as a last-resort safety net so that an unhandled exception here doesn't prevent the next polling operation. I think it should be the responsibility of the code that uses the HonoKafkaConsumer to make sure this doesn't happen, or to handle it properly, potentially also creating trace output.